examples/gen_test_data/gen_test_data/common.py

```diff
@@ -114,20 +118,28 @@ def print_progress(log_file_path: str, process):
                 line = f.readline().strip()
                 if line:
                     last_data_time = time.time()  # Update the time when the last data was received
-                    match = log_pattern.match(line)
-                    if not match:
+                    progress_match = progress_log_pattern.match(line)
+                    finished_match = finished_log_pattern.match(line)
+                    if not progress_match and not finished_match:
                         continue
 
-                    finished, total = map(int, match.groups())
-                    if progress_bar is None:
-                        progress_bar = tqdm(total=total, desc="Processing", file=sys.stdout)
-                    progress_bar.update(finished - progress_bar.n)
+                    if progress_match:
+                        finished, processing, pending = map(int, progress_match.groups())
+                        total = finished + processing + pending
+                        if progress_bar is None:
+                            progress_bar = tqdm(total=total, desc="Processing", miniters=1, file=sys.stdout)
+                            time.sleep(0.1)  # wait for 0.1 second to show the progress bar on next line
```

Comment:
due to tqdm issue, need to wait for 0.1 second between progress bar initialization and first update.

Reply:
updated to use mininterval